app/src/main/java/edu/grinnell/appdev/grinnelldirectory/APICaller.java

```diff
@@ -0,0 +1,138 @@
+
+import java.io.IOException;
+import java.util.List;
+
+import edu.grinnell.appdev.grinnelldirectory.Interfaces.APICallerInterface;
+import edu.grinnell.appdev.grinnelldirectory.Interfaces.DatabaseAPI;
+import edu.grinnell.appdev.grinnelldirectory.Model.Person;
+import edu.grinnell.appdev.grinnelldirectory.Model.User;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+
+
+public class APICaller {
+
+    private static final String BASE_URL = "https://itwebappstest.grinnell.edu/DotNet/WebServices/api/";
+
+    private Retrofit retrofit;
+    private DatabaseAPI dbAPI;
+    private Call<List<Person>> personQuery;
+    private APICallerInterface apiCallerInterface;
+    private User user;
+
+    public APICaller(User user1, APICallerInterface apiInterface) {
+        retrofit = new Retrofit.Builder().
+                baseUrl(BASE_URL).
+                addConverterFactory(GsonConverterFactory.create()).
+                build();
+        dbAPI = retrofit.create(DatabaseAPI.class);
+        apiCallerInterface = apiInterface;
+        user = user1;
+    }
+
+    /* - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -  */
+    public void simpleSearch(User user, List<String> fields) {
+        personQuery = dbAPI.simpleSearch(user, fields.get(0), fields.get(1), fields.get(2), fields.get(3));
+
+        personQuery.enqueue(new Callback<List<Person>>() {
+
+            @Override
+            public void onResponse(Call<List<Person>> call, Response<List<Person>> response) {
+                if (response.isSuccessful()) {
+                    Log.d("API_SIMPLE_SUCCESS", "API returned list of people.");
+                    // response.body() is the list of people, set to 'people'
+                    List<Person> people = response.body();
+                    apiCallerInterface.onSearchSuccess(people);
+                } else {
+                    try {
+                        Log.e("ERROR_SIMPLE_SEARCH", response.errorBody().string());
+                        apiCallerInterface.onServerFailure(response.raw().message());
+                    } catch (IOException e) {
+                        Log.e("API_FAILURE_EXCEPTION", e.toString());
+                        apiCallerInterface.onServerFailure(e.toString());
+                    }
+                }
+            }
+
+            @Override
+            public void onFailure(Call<List<Person>> call, Throwable t) {
+                Log.e("API_SIMPLE_FAILURE", t.toString());
+                apiCallerInterface.onNetworkingError(t.getMessage());
+            }
+        });
+    }
+
+    /* - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -  */
+    public void advancedSearch(User user, List<String> fields) {
+        personQuery = dbAPI.advancedSearch(user, fields.get(0), fields.get(1), fields.get(2)
```

Comment:
Another thing: We know for sure that chances are that `user` or `fields` will always return a valid value. But what happens in the case when either (or worse, both) returns `null`? We pass `user` as a parameter and access `fields`. As such, we'll immediately get a `NullPointerException` when we access the `fields`. In the case of `user`, we are passing around a null value (in the case of `user`) and eventually, we'll access user and reach a `NullPointerException`. 

Here's a potential "fix" (it won't fix it but it'll reasonably avert a potential app crash): 
Check to make sure that both `user` and `fields` are not `null`. If they are null, exit the function immediately via `return`. This will be a good time to think about how we can report potential crashes. 

This holds for every thing being returned by Retrofit. At best, Retrofit will just return a null and that's not good user experience, either. Since we don't know what Retrofit will do when the value for the key being retrieved is `null`, we need to be defensive. 

@prabirmsp @Samee24 : We should discuss Crashlytics or any better alternative.